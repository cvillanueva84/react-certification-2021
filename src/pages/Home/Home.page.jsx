import React from 'react';
import styled from 'styled-components';
import VideosList from '../../components/VideosList';

const Wrapper = styled.section`
  display: flex;
  flex-wrap: wrap;
  justify-content: space-between;
  padding: 2rem;
  margin-top: 70px;
  background-color: #fbe8d3;
  @media (max-width: 576px) {
    justify-content: center;
    padding: 1rem;
  }
`;

function HomePage() {
  return (
    <Wrapper>
      <VideosList />
    </Wrapper>
  );
}

export default HomePage;
