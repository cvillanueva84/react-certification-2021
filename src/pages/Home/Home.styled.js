import styled from 'styled-components';

const VideoCard = styled.div`
  background: #fff;
  box-shadow: 0px 2px 1px -1px rgba(0, 0, 0, 0.2), 0px 1px 1px 0px rgba(0, 0, 0, 0.14),
    0px 1px 3px 0px rgba(0, 0, 0, 0.12);
  border-radius: 4px;
  transition: all 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  width: 345px;
  margin: 15px;

  .videoTextContainer {
    padding: 16px;
  }
  .videoCardMedia {
    height: 140px;
    object-fit: cover;
    width: 100%;
  }
  .videoCardTitle {
    margin: 0;
    font-size: 1.25rem;
    font-family: 'Roboto', 'Helvetica', 'Arial', sans-serif;
    font-weight: 500;
    line-height: 1.6;
    letter-spacing: 0.0075em;
    margin-bottom: 0.35em;
  }
  .videoCardDescription {
    font-size: 0.875rem;
    font-family: 'Roboto', 'Helvetica', 'Arial', sans-serif;
    font-weight: 400;
    line-height: 1.43;
    letter-spacing: 0.01071em;
    color: rgba(0, 0, 0, 0.54);
  }
  &:hover {
    opacity: 0.75;
    cursor: pointer;
  }
`;

export { VideoCard };
