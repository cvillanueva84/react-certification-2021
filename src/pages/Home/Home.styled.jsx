import styled from 'styled-components';

export const Homepage = styled.section`
  width: 100%;
  text-align: center;
  justify-content: center;
  &:h2 {
    font-size: 3.75rem;
    font-weight: 300px;
    line-height: 1.2;
    font-family: 'Roboto', 'Helvetica';
  }
`;
