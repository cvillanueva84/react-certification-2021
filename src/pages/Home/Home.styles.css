.homepage {
  text-align: center;
}

.homepage h1 {
  color: #FFFFFF;
  font-size: 5rem;
  letter-spacing: -2px;
}
