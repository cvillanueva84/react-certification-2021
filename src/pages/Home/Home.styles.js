import styled from 'styled-components';

const StyledHome = styled.section`
  padding: 50px 100px;
  background-color: white;

  @media screen and (max-width: 768px) {
    padding: 30px 20px;
  }
`;

export { StyledHome }