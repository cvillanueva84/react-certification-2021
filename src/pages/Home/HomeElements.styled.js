import styled from 'styled-components';

export const HomeContainer = styled.section`
  max-width: 80rem;
  margin: 0 auto;
  width: 100%;

  h1 {
    text-align: center;
    font-size: 2rem;
    margin-bottom: 1.25rem;
    margin-bottom: 1.125rem;
  }
`;
