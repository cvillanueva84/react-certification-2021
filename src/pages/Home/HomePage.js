import React, { useContext, Suspense, lazy } from 'react';
import { SectionContainer, VideoListContainer, override } from './HomePage.styled';
import { VideoListContext } from '../../providers/VideoList/VideoList.provider';
import GridLoader from 'react-spinners/ClipLoader';
const VideoList = lazy(() => import('../../components/VideoList/VideoListComponent'));

const HomePage = () => {
  const { videosState } = useContext(VideoListContext);
  const { posts = [] } = videosState;

  return (
    <SectionContainer>
      <div>
        <h2 style={{ marginLeft: '4rem', marginTop: '2rem' }}>Our videos:</h2>
      </div>
      <Suspense fallback={<GridLoader size={150} css={override} />}>
        <VideoListContainer>
          {posts.map((video) => (
            <VideoList key={video.id.videoId} video={video} />
          ))}
        </VideoListContainer>
      </Suspense>
    </SectionContainer>
  );
};

export default HomePage;
