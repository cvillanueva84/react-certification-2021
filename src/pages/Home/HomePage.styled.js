import styled from 'styled-components'
import { css } from '@emotion/react';


export const SectionContainer = styled.section`
  width: 95%;
  margin: auto;
`;
SectionContainer.displayName = 'SectionContainer';
export const VideoListContainer = styled.div`
  display: flex;
  flex-wrap: wrap;
  justify-content: center;
`;
VideoListContainer.displayName = 'VideoListContainer';

export const override = css`
  display: block;
  margin: 20rem auto;
  border-color: var(--main-blue);
`;

