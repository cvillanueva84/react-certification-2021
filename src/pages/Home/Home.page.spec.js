import Home from './Home.page';

const { render, screen } = require('@testing-library/react');

jest.mock('../../components/VideosList/VideosList.component', () => {
  return function DummyVideosList() {
    return (
      <div data-testid="videos-list">
        <p>Dummy Videos List component</p>
      </div>
    );
  };
});

describe('Home Page', () => {
  let container;

  beforeEach(() => {
    container = render(<Home />).container;
  });

  test('Should renders Home page component', () => {
    expect(container).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });

  test('Should contains VideosList component', () => {
    const videosListsComponent = screen.getByTestId('videos-list');
    expect(videosListsComponent).toBeInTheDocument();
    expect(videosListsComponent.textContent).toBe('Dummy Videos List component');
  });
});
