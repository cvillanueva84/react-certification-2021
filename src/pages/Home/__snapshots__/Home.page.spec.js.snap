// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Page Should renders Home page component 1`] = `
<div>
  <section
    class="sc-bdnxRM iDdjcH"
  >
    <div
      data-testid="videos-list"
    >
      <p>
        Dummy Videos List component
      </p>
    </div>
  </section>
</div>
`;
